app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -222,12 +222,11 @@ private void sendAppViewLoadAnalytics(AppViewModel appViewModel) {
         .toObservable();
   }
 
-  private Observable<Boolean> loadBannerAds() {
+  private Observable<Boolean> loadBannerAds(AppModel appModel) {
     return appViewManager.shouldLoadBannerAd()
-        .observeOn(viewScheduler)
         .doOnSuccess(shouldLoadBanner -> {
           if (shouldLoadBanner) {
-            view.showBannerAd();
+            view.showBannerAd(appModel.isMature());
```

Comment:
this method only depends on the mature flag, we can remove the heavier dependency to not couple this method to the AppModel